addons/controllers/certificates_test.go

```diff
@@ -0,0 +1,118 @@
+package controllers
+
+import (
+	"path"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	adminregv1 "k8s.io/api/admissionregistration/v1"
+	cert2 "k8s.io/client-go/util/cert"
+	"k8s.io/client-go/util/keyutil"
+	"knative.dev/pkg/webhook/certificates/resources"
+
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/webhooks"
+)
+
+var _ = Describe("Webhook", func() {
+
+	Context("server's certificate and key", func() {
+		It("should be generated and written to the webhook server CertDir", func() {
+			Eventually(func() string { return webhookServer.CertDir }, "10s").ShouldNot(BeEmpty()) //Until the webhook server has comleted startup CertDir returns ""
+			certFilePath := path.Join(webhookServer.CertDir, webhookServer.CertName)
+			Expect(certFilePath).ShouldNot(BeEmpty())
+			keyFilePath := path.Join(webhookServer.CertDir, webhookServer.KeyName)
+			Expect(keyFilePath).ShouldNot(BeEmpty())
+			Expect(webhookServer).ToNot(BeNil())
+			secret, err := webhooks.UpdateTLSSecret(ctx, webhookScrtName, webhookServiceName, webhookServer, addonNamespace)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(secret).NotTo(BeNil())
+			cert, err := cert2.CertsFromFile(certFilePath)
```

Comment:
QQ - In this PR, do we have to create the certificates and keys manually for running tests ? I was trying to automate that as part of the tests so that when different components use webhooks they don't have to generate/rotate/manage certificates for integration tests.

Reply:
nope. 
In your test after the envtest has been created started and the manager has been created you call 
webhookserver = manager.getwebhook(), and then somewhere in your test you make the call: 

``` secret, err := webhooks.UpdateTLSSecret(ctx, webhookScrtName, webhookServiceName, webhookServer, addonNamespace)

you can see it here. 
https://github.com/vmware-tanzu/tanzu-framework/pull/1687/files#diff-03f91d8a3c07135ca5f572b41fb7e213ada82e2fc17fdac963a16388ee24af57R159

in this case the variable "webhookServer" is global and has a reference to the manager webhook server. See here:
https://github.com/vmware-tanzu/tanzu-framework/pull/1687/files#diff-03f91d8a3c07135ca5f572b41fb7e213ada82e2fc17fdac963a16388ee24af57R76